src/pbrt/cmd/imgtool.cpp

```diff
@@ -1854,11 +1854,45 @@ int makeenv(int argc, char *argv[]) {
             }
     });
 
+    // Upper hemisphere illuminance calculation for converting map to physical
```

Comment:
Doing it offline makes sense to me since all environment maps need to go through makeenv anyway (actually I'll need to add illuminance calculation to the makesky stuff too) in order to convert from latlong to equirect. It's a potentially expensive operation on large environment maps so makes sense to do it once and store it rather than every time you render.

That said we *could* do it in the ImageInfiniteLight Create() code path and that would have the advantage that it would be closer to the rest of the photometric code. Let me know what you prefer.

Reply:
They don't all need to go through makeenv, though: e.g. the ones in the pbrt-v4-scenes distribution are already in equiangular format, so really makeenv is really just for upgrading pbrt-v3 scenes (and is poorly named for that task..) So all in all, I'd lean toward doing it during Create()...

Reply:
OK this is done now. Only took 0.3s for an 8k image when I actually tested it so performance won't be a worry.